Public/Src/Tools/Xldb.Analyzer/XldbAnalyzer.cs

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Xldb.Proto;
+
+namespace Xldb.Analyzer
+{
+    /// <summary>
+    /// Xldb Analyzers that use the rocksDb instanceand the Xldb API to analyze BXL logs
+    /// </summary>
+    public class Program
+    {
+
+        private const string s_eventStatsAnalyzer = "eventstats";
+        private const string s_dumpPipAnalyzer = "dumppip";
+
+        private Dictionary<string, string> m_commandLineOptions = new Dictionary<string, string>();

```

Comment:
You might want to pass `StringComparer.OrdinalIgnoreCase`

that way `/m` and `/M` work both

Reply:
I convert everything to lower in ProcessArguments.

---
In reply to: [315939713](https://github.com/microsoft/BuildXL/pull/775#discussion_r315939713) [](ancestors = 315939713)